docs/BacwardCompatibility.md

```diff
@@ -0,0 +1,133 @@
+Before talking about how to provide compatability, it is worthwhile to clarify what we mean by compatibility. 
+Fybrik is required to provide forwards and backwards compatibility in its behaviour. Compatibility is hard, especially 
+handling issues around rollback-safety. 
+Let's try to define compatability in this way:
+- Existing clients need not be aware of a new change in order to continue to function as they did previously, 
+even when the change is in use.
+- Any Fybrik operation that succeeded before the new change must succeed after the change.
+- Any Fybrik operation that does not use the new change must behave the same as it did before the change.
+- Any Fybrik operation that use the new change must not cause problems (e.g. crash or degrade behavior) when issued 
+against Fybrik objects, e.g. FybrikModules, Plotters, Blueprints and FybrikApplications created before the functionality was added.
+The objects should be smoothly promoted to the new version (if required).
+- It must be possible to round-trip Fybrik version changes (convert to different API versions and back) with no loss of 
+information (Should we support downgrade ?)
+
+## Fybrik Version Policy
+Fybrik uses [semantic versioning model](https://semver.org), which is the module adopted by a wide range of projects,
+e.g. Kubernetes, goLang, Helm and many others.
+
+Fybrik versions are of the form "vX.Y.Z" where X is the major version number, Y is the minor version number
+and Z is the patch release number.
+- Major Releases - released less frequently, indicate that **backward-incompatible public API changes**. This release
+  carries no guarantee that it will be backward compatible with preceding major versions.
+- Minor Releases - indicate **backward-compatible public API changes**. This release guarantees backward compatibility
+  and stability. The backward compatibility policy is defined in a separate document [ADD link] but in general, Fybrik
+  supports 2 previous minor releases.
+- Patch Releases - should not change main functionality, mostly for security or bug fixes, and therefore guarantee
+  backward compatibility.
+
+*Note:* Kubernetes uses similar semantic version model, but its versions do not have the "v" prefix. In order to use
+the same release tag and goLang module tag, we have chosen goLang releases modules pattern with the "v" prefix.
+
+### Backward compatibility policy
+Fybrik will support up to 2 backward minor releases, e.g. the version v1.5.x should be able to work with releases v1.4.x
+and v1.3.x
+
+## Fybrik components or features that can influence compatability 
+
+*Note:* the list is not closed, and can be extended
+
+- [Kubernetes Custom Resources](#kubernetes-custom-resource-versions-and-upgrade-plan)
+- [Data Catalog Connector Interface](#data-catalog-and-policy-manager-connector-interfaces)
+- [Policy Manager Interface](#data-catalog-and-policy-manager-connector-interfaces)
+- IT Configuration Policies
+- IT attributes
+- ....
```

Comment:
Include links to the interface definitions

Reply:
OK

Reply:
I included the links in each sub-paragraph. Do you think the links should be in the content list here ?